lib/SILGen/SILGenExpr.cpp

```diff
@@ -3081,7 +3081,13 @@ visitObjectLiteralExpr(ObjectLiteralExpr *E, SGFContext C) {
 
     // Emit the tensor arguments as well as the attribute values.
     for (auto &elt : tuple->getElements().drop_front()) {
-      args.push_back(visit(elt).forwardAsSingleValue(SGF, elt));
+      // Arguments of this builtin will be taken at +0 instead of +1, for
+      // consistency with the default calling convention of taking function
+      // parameters as @guaranteed instead of @owned.
+      // e.g. Say E represents #tfop("Add", x, x). x can be passed into this
+      // expression without having to do a strong_retain (or copy_value) first.
+      args.push_back(visit(elt, SGFContext(SGFContext::AllowImmediatePlusZero))
```

Comment:
I see a lot of code in SILGen using a "emitRValueAsSingleValue" helper.  Would it make sense to use it?

Reply:
Actually, I don't think that's correct.  I think we have to use the "AllowGuaranteedPlusZero" form of this, because we take multiple operands and they can be arbitrarily complicated.  Does this work?

emitRValueAsSingleValue(elt, SGFContext::AllowGuaranteedPlusZero);

Reply:
Done.

Also replaced the call of forwardAsSingleValue() with getValue(), since the tfop builtin no longer consumes the arg at +1, so we should not disable the cleanup of the arg.

Also changed AllowImmediatePlusZero to AllowGuaranteedPlusZero, though from code reading i don't see any call-sites using that value. Am I missing something? I can start a thread on swift compiler forum to clarify this if needed.